src/http.cc

```diff
@@ -1790,7 +1790,7 @@ HttpStateData::doneWithServer() const
  * Fixup authentication request headers for special cases
  */
 static void
-httpFixupAuthentication(HttpRequest * request, const HttpHeader * hdr_in, HttpHeader * hdr_out, const Http::StateFlags &flags)
+httpFixupAuthentication(HttpRequest * request, const HttpHeader * hdr_in, HttpHeader * hdr_out, char *peerHost, const Http::StateFlags &flags)
```

Comment:
To simplify (some) callers and to reduce the number of places where we need to decide what "we are talking to a cache_peer" really means, let's pass a CachePeer pointer here instead of a c-string. If you can pass a pointer to a _constant_ CachePeer, please do that, of course.

Reply:
Done.